core/types/calls.cc

```diff
@@ -3010,6 +3010,31 @@ class Module_tripleEq : public IntrinsicMethod {
     }
 } Module_tripleEq;
 
+class T_Enum_tripleEq : public IntrinsicMethod {
+public:
+    void apply(const GlobalState &gs, const DispatchArgs &args, DispatchResult &res) const override {
+        if (args.args.size() != 1) {
+            return;
+        }
+        auto rhs = args.args[0]->type;
+        if (rhs.isUntyped()) {
+            res.returnType = rhs;
+            return;
+        }
+        auto lhs = args.thisType;
+        ENFORCE(!lhs.isUntyped(), "lhs of T::Enum.=== must be typed");
+        if (Types::isSubType(gs, rhs, lhs)) {
+            res.returnType = Types::trueClass();
+            return;
+        }
+        if (Types::glb(gs, rhs, lhs).isBottom()) {
+            res.returnType = Types::falseClass();
```

Comment:
Can you make sure to write a test for this?

I think that this should work:

```ruby
T.reveal_type(MyEnum::Y.===(MyEnum::X))
```

also it would be great to have both the examples from #4115 committed as test cases.